nh_clinical/activity_extension.py

```diff
@@ -538,110 +530,241 @@ def get_activity_user_ids(self, cr, uid, activity_id, context=None):
+        supplied policy dict
+
+        :param activity_id: ID of the activity to change state of
+        :param policy_dict: Dictionary representing the policy to enact on the
+            activity
+        """
+        activity_model = self.env['nh.activity']
+        activity = activity_model.browse(activity_id)
+        state = policy_dict.get('type')
+        if state == 'start':
+            activity.sudo(SUPERUSER_ID).start()
+        elif state == 'complete':
+            data = policy_dict.get('data')
+            if data:
+                activity.sudo(SUPERUSER_ID).submit(data)
+            activity.sudo(SUPERUSER_ID).complete()
+        else:
+            recurring = state == 'recurring'
+            schedule_date = self.get_schedule_date(
+                activity, recurring=recurring)
+            activity.sudo(SUPERUSER_ID).schedule(schedule_date)
+
+    @api.model
+    def trigger_policy(self, activity_id, location_id=None, case=False):
         """
         Triggers the list of activities in the ``_POLICY['activities']``
         list.
 
-        :param activity_id: id of activity triggering policy
+        :param activity_id: ID of the activity associated with the
+            record
         :type activity_id: int
         :param location_id: location id [optional].
             Required for checking context
         :type location_id: int
         :param case: default ``False``. Otherwise integer related to
             risk of patient
         :type case: bool or int
-        :returns: ``True``
+        :returns: ``False`` if no spell activity otherwise ``True``
         :rtype: bool
         """
-        activity_pool = self.pool['nh.activity']
-        spell_pool = self.pool['nh.clinical.spell']
-        location_pool = self.pool['nh.clinical.location']
-        if self._POLICY.get('activities', []):
-            activity = activity_pool.browse(cr, SUPERUSER_ID, activity_id,
-                                            context)
-            spell_id = spell_pool.get_by_patient_id(
-                cr, SUPERUSER_ID, activity.data_ref.patient_id.id,
-                context=context)
-            if spell_id:
-                spell_activity_id = spell_pool.browse(
-                    cr, uid, spell_id, context=context).activity_id.id
-            else:
-                return False
-
-        else:
-            return True
+        # If no spell associated with activity then we want to just return
+        # TODO: Refactor this so instead of passing over an activity ID it
+        # can just load a record.
+        activity_model = self.env['nh.activity']
+        activity = activity_model.browse(activity_id)
+        spell_activity = activity_model.search(
+            [
+                ['state', 'not in', ['completed', 'cancelled']],
+                ['patient_id', '=', activity.patient_id.id],
+                ['data_model', '=', 'nh.clinical.spell']
+            ]
+        )
+        if not spell_activity.id:
```

Comment:
No need for the `.id` because the recordset itself will be falsey if nothing was found.

Reply:
👍